### PR TITLE
chore: add lsp clean-ns in lint-fix

### PR DESCRIPTION
**Pull request by yqrashawn**
-------

devtool change only

`make lint-fix` will run lsp `clean-ns` after this PR

status: ready

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -325,6 +325,7 @@ lint-fix: ##@test Run code style checks and fix issues
 	ALL_CLOJURE_FILES=$(call find_all_clojure_files) && \
 	zprint '{:search-config? true}' -sw $$ALL_CLOJURE_FILES && \
 	zprint '{:search-config? true}' -sw $$ALL_CLOJURE_FILES && \
+	clojure-lsp --ns-exclude-regex ".*/src/status_im2/core\.cljs$$" clean-ns && \
```
Comment (yqrashawn) [inline Makefile:327]: `--ns-exclude-regex` is not what it looks like
it's actually lint against file URL like this `file:///Users/yqrashawn/workspace/office/status-mobile/src/status_im/ui/screens/profile/contact/views.cljs`

should create an issue for clojure-lsp

**Review by flexsurfer — APPROVED** [review]

File: .lsp/config.edn [hunk 2]
```diff
@@ -0,0 +1 @@
+{:clean {:ns-inner-blocks-indentation :keep}}
```
Comment (yqrashawn) [inline .lsp/config.edn:1]: find this in one clojure-lsp issue https://github.com/clojure-lsp/clojure-lsp/issues/1141
yet it's not in clojure-lsp's [all-available-settings.edn](https://github.com/clojure-lsp/clojure-lsp/blob/master/docs/all-available-settings.edn)

should create an issue for clojure-lsp

**Review by ilmotta — APPROVED** [review]

File: .lsp/config.edn (hunk 2 above)
Comment (ilmotta) [inline .lsp/config.edn:1]: @yqrashawn, I haven't checked, but do you know if we can add a setting to the edn file excluding certain namespaces? Just so nobody sorts the `status-im2.core` namespace by accident in their text editors.

File: nix/shell.nix [hunk 3]
```diff
@@ -23,7 +23,7 @@ in mkShell {
       # build specific utilities
       clojure maven watchman
       # lint specific utilities
-      clj-kondo zprint
+      clj-kondo zprint clojure-lsp
```
Comment (jakubgs) [inline nix/shell.nix]: The `lint` and `lint-fix` targets use the `clojure` shell, so we should probably just put this in there. Honestly, the other 2 utilities here should probably also be in the `clojure` shell only:
https://github.com/status-im/status-mobile/blob/fc284f466fff16790fbd6c2585f54f2bb1209c54/nix/shells.nix#L19-L29

**Review by jakubgs — CHANGES_REQUESTED** [review]
Please move `clj-kondo`, `zprint`, and `clojure-lsp` to `clojure` nix shell.

It's better to use smaller per-purpose shells like the `clojure` one so its startup is faster.

File: .lsp/config.edn (hunk 2 above)
Comment (yqrashawn) [reply .lsp/config.edn:1]: Took a brief look at clojure-lsp's source code. I don't think it's possible. I guess we can at least add checks with clj-kondo for `status-im2.core`

added to the lint issue https://github.com/status-im/status-mobile/issues/17641

**Comment by yqrashawn** [comment]
@jakubgs  fixed

**Review by jakubgs — APPROVED** [review]
Much better.

**Pull request merged**